### PR TITLE
Property in JsonResponseHandler to allow format compliant of the json:api spec

### PR DESCRIPTION
**Pull request by acidjazz**
Addresses issue #442 , call the method`setJsonApi()` of the `JsonResponseHandler` passing in `true` and receive a result compliant to the [json:api specification](http://jsonapi.org/examples/#error-objects)

```json
{
  "errors": [
    {
      "type": "ErrorException",
      "message": "Undefined variable: bob",
      "file": "/Users/k/basal/app/Api/UserController.php",
      "line": 28
    }
  ]
}
```

### CODE REVIEW

**Comment by prisis** [comment]
You need to remove the fixes for prettyprint, this pr should only have changes for the json api fix

File: src/Whoops/Handler/JsonResponseHandler.php [hunk 1]
```diff
@@ -21,6 +21,14 @@ class JsonResponseHandler extends Handler
     private $returnFrames = false;
 
     /**
+     * return the json:api compliant version
+     * @var bool
+     */
+    public $jsonApi = false;
```
Comment (prisis) [inline src/Whoops/Handler/JsonResponseHandler.php]: Pls use a setter and make this protected or private

File: src/Whoops/Handler/JsonResponseHandler.php [hunk 2]
```diff
@@ -21,6 +21,14 @@ class JsonResponseHandler extends Handler
...
+
+
+
```
Comment (prisis) [inline src/Whoops/Handler/JsonResponseHandler.php]: Remove this line.

**Comment by acidjazz** [comment]
@prisis I thought of that but being as those are more stable than the previous PR that was merged, I included them, if for some reason that PR is denied I'll happily re-create this one sans-fixes

File: src/Whoops/Handler/JsonResponseHandler.php (hunk 2 above)
Comment (acidjazz) [inline src/Whoops/Handler/JsonResponseHandler.php]: done

**Comment by prisis** [comment]
We follow one pr for one fix

File: src/Whoops/Handler/JsonResponseHandler.php (hunk 1 above)
Comment (acidjazz) [inline src/Whoops/Handler/JsonResponseHandler.php]: done

**Comment by acidjazz** [comment]
OK well lets follow this one after #439

File: src/Whoops/Handler/JsonResponseHandler.php [hunk 3]
```diff
@@ -21,6 +21,26 @@ class JsonResponseHandler extends Handler
...
+    public function setJsonApi($jsonApi=false)
+    {
+
+      if (is_bool($jsonApi)) {
+        $this->jsonApi = $jsonApi;
+        return true;
+      }
+
+      return false;
+    }
```
Comment (prisis) [inline src/Whoops/Handler/JsonResponseHandler.php]: /**
     * @param bool Default is false
     * @return $this
     */
    public function setJsonApi($jsonApi = false)
    {

        $this->jsonApi = (bool) $jsonApi;

        return $this;
    }
is less code and do the same thing

File: src/Whoops/Handler/JsonResponseHandler.php [hunk 4]
```diff
@@ -39,12 +59,24 @@ public function addTraceToOutput($returnFrames = null)
      */
     public function handle()
     {
+
+      if ($this->jsonApi == true) {
```
Comment (prisis) [inline src/Whoops/Handler/JsonResponseHandler.php]: Use === not ==

Comment (acidjazz) [inline src/Whoops/Handler/JsonResponseHandler.php]: done

File: src/Whoops/Handler/JsonResponseHandler.php (hunk 3 above)
Comment (acidjazz) [inline src/Whoops/Handler/JsonResponseHandler.php]: done

File: docs/API Documentation.md [hunk 5]
```diff
@@ -279,7 +279,7 @@ CallbackHandler::handle()
 
 ## <a name="handler-json"></a> `Whoops\Handler\JsonResponseHandler`
 
-The `JsonResponseHandler`, upon receiving an exception to handle, simply constructs a `JSON` payload, and outputs it. Methods are available to control the detail of the output, and if it should only execute for AJAX requests - paired with another handler under it, such as the `PrettyPageHandler`, it allows you to have meaningful output for both regular and AJAX requests. Neat!
+The `JsonResponseHandler`, upon receiving an exception to handle, simply constructs a `JSON` payload, and outputs it. Methods are available to control the detail of the output, and if it should only execute for AJAX requests - paired with another handler under it, such as the `PrettyPageHandler`, it allows you to have meaningful output for both regular and AJAX requests. A property is availble to alter the style of output as well. Neat!
```
Comment (staabm) [inline docs/API Documentation.md]: This handler optionally supports the  jsonapi spec.

File: examples/example-ajax-only.php [hunk 6]
```diff
@@ -37,6 +37,11 @@
     // You can also tell JsonResponseHandler to give you a full stack trace:
     // $jsonHandler->addTraceToOutput(true);
 
+    // You can also return a result compliant to the json:api spec
+    // re: http://jsonapi.org/examples/#error-objects
+    // tl;dr: error[] becomes errors[[]]
```
Comment (staabm) [inline examples/example-ajax-only.php]: Drop this line

File: examples/example-ajax-only.php [hunk 7]
```diff
@@ -37,6 +37,11 @@
...
+   $jsonHandler->setJsonApi(true);;
```
Comment (staabm) [inline examples/example-ajax-only.php]: 2 semicolons

File: src/Whoops/Handler/JsonResponseHandler.php [hunk 8]
```diff
@@ -21,6 +21,21 @@ class JsonResponseHandler extends Handler
     private $returnFrames = false;
 
     /**
+     * @var bool
+     */
+    private $jsonApi = false;
+
+    /**
+     * @param bool Default is false
+     * @return $this
```
Comment (staabm) [inline src/Whoops/Handler/JsonResponseHandler.php]: Needs some description

**Comment by denis-sokolov** [comment]
I think we can accept this. Could you please rebase and ensure the diff is about the json api spec? At the moment in my diff view I see prettyprint related changes.

**Comment by acidjazz** [comment]
@denis-sokolov where should i start? before the prettyprint merge?

**Pull request closed without merging**